src/components/KPrompt/KPrompt.vue

```diff
@@ -1,332 +1,189 @@
 <template>
   <KModal
     class="k-prompt"
-    :is-visible="isVisible"
-    :tabbable-options="tabbableOptions"
-    text-align="left"
-    :title="displayTitle"
+    :title="title || 'Confirm your action'"
+    :visible="visible"
+    v-bind="{ ...sanitizedAttrs, ...modalAttributes}"
+    @cancel="$emit('cancel')"
+    @proceed="$emit('proceed')"
   >
-    <template #header-content>
-      <div class="k-prompt-header">
-        <div class="k-prompt-header-content">
-          <slot name="header-content">
-            <KIcon
-              v-if="type === 'warning'"
-              class="warning-icon"
-              :color="`var(--kui-color-text-inverse, ${KUI_COLOR_TEXT_INVERSE})`"
-              icon="warning"
-              secondary-color="currentColor"
-              :size="KUI_ICON_SIZE_40"
-            />
-            {{ displayTitle }}
-          </slot>
-          <div class="close-button">
-            <KButton
-              aria-label="Close"
-              @click="close"
-            >
-              <KIcon
-                :color="`var(--kui-color-text-neutral, ${KUI_COLOR_TEXT_NEUTRAL})`"
-                icon="close"
-                :size="KUI_ICON_SIZE_30"
-              />
-            </KButton>
-          </div>
-        </div>
-        <hr class="divider">
-      </div>
+    <template
+      v-if="$slots.title"
+      #title
+    >
+      <slot name="title" />
     </template>
-    <template #body-content>
-      <div class="k-prompt-body">
-        <div class="k-prompt-body-content">
-          <slot name="body-content">
+    <template #default>
+      <div
+        v-if="$slots.default || message"
+        class="prompt-content"
+      >
+        <slot name="default">
+          <p class="prompt-message">
             {{ message }}
-          </slot>
-
-          <div
-            v-if="confirmationText"
-            class="k-prompt-confirm-text"
-          >
-            Type "<span class="confirm-text">{{ confirmationText }}</span>" to confirm your action.
+          </p>
+        </slot>
+      </div>
 
-            <KInput
-              v-model="confirmationInput"
-              autocapitalize="off"
-              autocomplete="off"
-              data-testid="confirmation-input"
-            />
-          </div>
-        </div>
-        <hr class="divider">
+      <div
+        v-if="confirmationText"
+        class="prompt-confirmation-container"
+      >
+        <p class="prompt-confirmation-text">
+          {{ confirmationPromptText[0] }} <span class="confirmation-text">"{{ confirmationText }}"</span>
+          {{ confirmationPromptText[1] ? confirmationPromptText[1] : '' }}
+        </p>
+        <KInput
+          v-model.trim="confirmationInput"
```

Comment:
issue: do not trim here; they **must** type the exact text
```suggestion
          v-model="confirmationInput"
```